### PR TITLE
fix: Fix tables creation depending on each other via foreignKey constraint

### PR DESCRIPTION
**Pull request by naftalmm**
This PR fixes https://github.com/JetBrains/Exposed/issues/1643

### CODE REVIEW

**Review by AlexeySoshin — APPROVED** [review]

**Review by joc-a — COMMENTED** [review]
Hi @naftalmm. Thank you for your contribution. Could you please rebase on the `main` branch?

File: exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/CreateMissingTablesAndColumnsTests.kt [hunk 1]
```diff
@@ -620,7 +620,7 @@ class CreateMissingTablesAndColumnsTests : DatabaseTestsBase() {
 
     @Test
     fun testCreateCompositePrimaryKeyTableAndCompositeForeignKeyTableMultipleTimes() {
-        withTables(CompositePrimaryKeyTable, CompositeForeignKeyTable) {
+        withTables(CompositeForeignKeyTable, CompositePrimaryKeyTable) {
```
Comment (e5l) [inline exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/CreateMissingTablesAndColumnsTests.kt:623]: I would make a separate test to check the order: it should be correct in any

**Review by e5l — APPROVED** [review]
Thanks for the contribution, lgtm! please consider adding a test

**Comment by joc-a** [comment]
@naftalmm Thanks for adding tests. Could you please rebase on the `main` branch and resolve conflicts?

**Review by joc-a — APPROVED** [review]

**Pull request merged**